cpu/atmega_common/periph/uart.c

```diff
@@ -121,6 +126,8 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
         return UART_NODEV;
     }
 
+    while (_tx_pending) {}
```

Comment:
Would it make sense to add a timeout ? This would avoid a potential endless loop here. Or maybe `_tx_pending` is also cleared at some point.

```suggestion
    uint16_t count = 0xffff;
    while (_tx_pending || count--) {}
```

Reply:
It would need to be while (_tx_pending && count--), and a `16bit` counter is not enough for `arduino-uno`

Reply:
My bad, it does, it was just optimized out when alone.

Reply:
BTW:

```
This flag bit is set when the entire frame in the Transmit Shift Register has been shifted out andthere are no new data currently present in the transmit buffer (UDRn). The TXCn Flag bit is auto-matically cleared when a transmit complete interrupt is executed, or it can be cleared by writinga one to its bit location. The TXCn Flag can generate a Transmit Complete interrupt (seedescription of the TXCIEn bit)
```

So this could also work without ISR, just checking the status of `TXCn Flag` and with a timeout. Do you have a prefernece?

Reply:
I think the timeout is a good idea in case the ISR is somehow missed.

Reply:
And ok for the `&&` (was not completely awake when I wrote this...)